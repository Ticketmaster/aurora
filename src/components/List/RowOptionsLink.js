import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import classnames from "classnames";

import { Column, Row } from "../Grid";
import { LinkDeprecated } from "../Text";
import { mediumAndUp } from "../../theme/mediaQueries";
import constants from "../../theme/constants";
import spacing from "../../theme/spacing";

import {
  CHEVRON_ICON_PADDING,
  CHEVRON_ICON_SIZE,
  ROW_DATE_MEDIUM_WIDTH,
  ROW_DATE_SMALL_WIDTH
} from "./constants";

const Content = styled(Row)`
  width: 100%;
  padding-left: ${ROW_DATE_SMALL_WIDTH};
  padding-bottom: ${spacing.cozy};
  ${mediumAndUp`
    padding-left: calc(${CHEVRON_ICON_SIZE}px + ${CHEVRON_ICON_PADDING} + ${CHEVRON_ICON_PADDING} + ${ROW_DATE_MEDIUM_WIDTH});
  `};
`;

const MobileLink = styled(LinkDeprecated)`
  display: inline-block;
  ${mediumAndUp`
    display: none;
  `};
`;

const DesktopLink = styled(LinkDeprecated)`
  display: none;
  ${mediumAndUp`
    display: inline-block;
    opacity 1;
    line-height: 1.5;
    transition: opacity 0.3s ${constants.easing.easeInOutQuad} 0.2s;
    &.link--hidden {
      opacity 0;
      transition: opacity 0.1s ${constants.easing.easeInQuad};
    }
  `};
`;

const RowOptionsLink = ({ variant, isOpen, url, index, onClick, children }) =>
  variant === "withLink" && (
    <Content>
      <Column>
        <MobileLink
          linkUrl={url}
          data-index={index}
          onClick={onClick}
          className="link--row-options"
        >
          {children}
        </MobileLink>
        <DesktopLink
          href={url}
          data-index={index}
          className={classnames({
            "link--row-options": !url.length,
            "link--hidden": isOpen
          })}
        >
          {children}
        </DesktopLink>
      </Column>
    </Content>
  );

RowOptionsLink.defaultProps = {
  index: 0,
  isOpen: false,
  variant: "",
  url: "",
  onClick: () => {},
  children: null
};

RowOptionsLink.propTypes = {
  index: PropTypes.number,
  isOpen: PropTypes.bool,
  variant: PropTypes.string,
  url: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node
};

export default RowOptionsLink;
