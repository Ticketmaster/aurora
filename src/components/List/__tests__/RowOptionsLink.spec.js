import React from "react";
import renderer from "react-test-renderer";
import RowOptionsLink from "../RowOptionsLink";

describe("<RowOptionsLink />", () => {
  it("renders RowOptionsLink", () => {
    const component = renderer.create(<RowOptionsLink variant="withLink" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
