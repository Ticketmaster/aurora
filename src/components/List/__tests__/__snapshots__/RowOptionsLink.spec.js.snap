// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowOptionsLink /> renders RowOptionsLink 1`] = `
.c1 {
  box-sizing: border-box;
  padding-right: 8px;
  padding-left: 8px;
  max-width: 100%;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  text-transform: none;
  color: rgba(2,108,223,1);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.3s ease;
  transition: color 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  outline: 0;
  background: transparent;
  padding: 0;
  cursor: pointer;
}

.c3:focus,
.c3:active,
.c3:visited,
.c3:hover {
  color: rgba(2,108,223,1);
}

.c3:hover {
  color: #0150a7;
}

.c5 {
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  text-transform: none;
  color: rgba(2,108,223,1);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.3s ease;
  transition: color 0.3s ease;
  cursor: pointer;
}

.c5:focus,
.c5:active,
.c5:visited,
.c5:hover {
  color: rgba(2,108,223,1);
}

.c5:hover {
  color: #0150a7;
}

.c0 {
  width: 100%;
  position: absolute;
  left: 0;
  top: 100%;
}

.c2 {
  display: inline-block;
}

.c4 {
  display: none;
}

@media screen and (min-width:768px) {
  .c1 {
    max-width: 100%;
    padding-right: 12px;
    padding-left: 12px;
    -webkit-flex: 0 0 100%;
    -ms-flex: 0 0 100%;
    flex: 0 0 100%;
  }
}

@media screen and (min-width:1024px) {
  .c1 {
    max-width: 100%;
    -webkit-flex: 0 0 100%;
    -ms-flex: 0 0 100%;
    flex: 0 0 100%;
  }
}

@media screen and (min-width:1440px) {
  .c1 {
    max-width: 100%;
    -webkit-flex: 0 0 100%;
    -ms-flex: 0 0 100%;
    flex: 0 0 100%;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    font-size: 14px;
  }
}

@media screen and (min-width:1024px) {
  .c3 {
    font-size: 14px;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    font-size: 14px;
  }
}

@media screen and (min-width:1024px) {
  .c5 {
    font-size: 14px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c4 {
    display: inline-block;
    opacity: 1;
    line-height: 1.5;
    -webkit-transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955) 0.2s;
    transition: opacity 0.3s cubic-bezier(0.455,0.03,0.515,0.955) 0.2s;
  }

  .c4.link--hidden {
    opacity: 0;
    -webkit-transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
    transition: opacity 0.1s cubic-bezier(0.55,0.085,0.68,0.53);
  }
}

<div
  className="c0 c1"
>
  <button
    className="link--row-options c2 c3"
    data-index={0}
    href={null}
    onClick={[Function]}
    rel="_self"
    size={
      Object {
        "large": "hecto",
        "medium": "hecto",
        "small": "hecto",
      }
    }
    target=""
  />
  <span
    className="link--row-options c4 c5"
    data-index={0}
    href=""
    onClick={null}
    rel="_self"
    size={
      Object {
        "large": "hecto",
        "medium": "hecto",
        "small": "hecto",
      }
    }
    target=""
  />
</div>
`;
